drift/lib/src/runtime/manager/annotate.dart

```diff
@@ -0,0 +1,92 @@
+part of 'manager.dart';
+
+/// A class that contains the information needed to create an annotation
+sealed class BaseAnnotation<SqlType extends Object, $Table extends Table> {
+  /// The expression/column which will be added to the query
+  Expression<SqlType> get _expression;
+
+  /// The join builders that are needed to read the expression
+  final Set<JoinBuilder> _joinBuilders;
+  BaseAnnotation(this._joinBuilders);
+
+  /// Create an order by clause for this annotation
+  ColumnOrderings get order {
+    return ColumnOrderings(_expression);
+  }
+}
+
+/// A class that contains the information needed to create an annotation
```

Comment:
This comment explains nothing. It should mention what an annotation is (essentially a manager wrapper around arbitrary SQL expressions with reference information resolved), probably link to the `manager/#annotation` site for details and explain why this is distinct from `BaseAnnotation` / only for columns without converters.

I know it feels tedious to write dartdocs when the main documentation is available and provides more context, but documentation comments still serve as an entrypoint for people interested in more details or those studying the implementation. So they should provide some self-contained information.

Reply:
Great point. Updated